lib/Controller/DocumentController.php

```diff
@@ -160,11 +169,53 @@ private function domainOnly($url) {
 		return "$scheme$host$port";
 	}
 
+	/**
+	 * Redirect to the files app with proper CSP headers set for federated editing
+	 *
+	 * This is a workaround since we cannot set a nonce for allowing dynamic URLs in the richdocument iframe
+	 *
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function open($fileId) {
+		try {
+			$folder = $this->rootFolder->getUserFolder($this->uid);
+			$item = $folder->getById($fileId)[0];
+			if (!($item instanceof Node)) {
+				throw new \Exception();
+			}
+
+			if ($item->getStorage()->instanceOfStorage(\OCA\Files_Sharing\External\Storage::class)) {
+				$remote = $item->getStorage()->getRemote();
+				$remoteCollabora = $this->federationService->getRemoteCollaboraURL($remote);
+				if ($remoteCollabora !== '') {
+					$fullPath = explode('/', $item->getPath());
+					$dir = implode('/', array_slice($fullPath, 3, -1));
```

Comment:
@rullzer Is there a better way to get the path of the node inside of the user folder? getInternalPath() returns the path starting from inside of the storage only

Reply:
On a file, get the absolute path. And then do a getrelative path with that on the userfolder

Reply:
Thanks, works like a charm.